### PR TITLE
Streamline Noid::Minter#dump to the essentials

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

**Comment by mjgiarlo** [comment]
I'll take a look tomorrow. Thanks for this, @cbeer

File: lib/noid/minter.rb [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
-      seed(options[:seed], options[:seq])
+
+      @counters = options[:counters]
+      @max_counters = options[:max_counters]
+
+      # callback when an identifier is minted
+      @after_mint = options[:after_mint]
+
+      # used for random minters
+      @rand = options[:rand] if options[:rand].is_a? Random
```
Comment (mjgiarlo) [inline lib/noid/minter.rb]: These three lines hurt me a bit. This unpacks the path through the lines more clearly to me:

```ruby
@rand = case options[:rand]
        when Random
          options[:rand]
        when String
          Marshal.load(options[:rand])
        else
          Random.new(options[:seed] || Random.new_seed)
        end
end
```

File: lib/noid/minter.rb [hunk 2]
```diff
@@ -93,7 +95,16 @@ def counters
     end
 
     def dump
-      { state: true, seq: @seq, seed: @seed, template: template.template, counters: Marshal.load(Marshal.dump(counters)) }
+      {
+        template: template.template,
+        counters: Marshal.load(Marshal.dump(counters)),
```
Comment (mjgiarlo) [inline lib/noid/minter.rb]: Why the load here?

Comment (cbeer) [inline lib/noid/minter.rb]: Because we were `load`ing before. I think the idea was to deep dup the counters.

Comment (mjgiarlo) [inline lib/noid/minter.rb]: Ah, the line scrolled off the screen and I missed that this is preserving old behavior. Moving right along.

**Pull request merged**